pallets/frequency-tx-payment/src/benchmarking.rs

```diff
@@ -1,17 +1,54 @@
 use super::*;
-
+use common_primitives::msa::MessageSourceId;
 use frame_benchmarking::v2::*;
-use frame_system::{Call as SystemCall, RawOrigin};
+use frame_support::assert_ok;
+use frame_system::RawOrigin;
+use pallet_msa::AddProvider;
+use sp_core::{sr25519, Pair};
+use sp_runtime::{
+	traits::{AsTransactionAuthorizedOrigin, DispatchTransaction},
+	MultiSignature,
+};
+
+pub fn register_provider<T: Config>(target_id: MessageSourceId, name: &'static str) {
+	#[allow(clippy::useless_conversion)]
+	let name = Vec::from(name).try_into().expect("error");
+	assert_ok!(T::ProviderBenchmarkHelper::create(target_id, name));
+}
 
-#[benchmarks]
+pub fn register_specified_msa<T: Config>(target_id: MessageSourceId, account: T::AccountId) {
+	assert_ok!(T::MsaBenchmarkHelper::create_msa(target_id, account.clone()));
```

Comment:
Similar here, we can call `Msa::<T>::create_account()` instead of injecting the MsaBenchmarkHelper.